egs2/librispeech/hubert_asr1/hubert_asr.sh

```diff
@@ -0,0 +1,552 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+min() {
+  local a b
+  a=$1
+  for b in "$@"; do
+      if [ "${b}" -le "${a}" ]; then
+          a="${b}"
+      fi
+  done
+  echo "${a}"
+}
+SECONDS=0
+
+# General configuration
+stage=1              # Processes starts from the specified stage.
+stop_stage=10000     # Processes is stopped at the specified stage.
+pretrain_start_iter= # Pretrain starts from the specified iteration (0 mean MFCC iteraion)
+pretrain_stop_iter=  # Pretrain is stopped from the specified iteration (0 mean MFCC iteraion)
+skip_data_prep=false # Skip data preparation stages.
+skip_eval=false      # Skip decoding and evaluation stages.
+skip_upload=true     # Skip packing and uploading stages.
+pretrain_ngpu=      # The number of gpus in pretrain stage ("0" uses cpu, otherwise use gpu).
+pretrain_num_nodes=1 # The number of nodes in pretrain stage.
+finetune_ngpu=      # The number of gpus in finetune stage("0" uses cpu, otherwise use gpu).
+finetune_num_nodes=1 # The number of nodes in finetune stage.
```

Comment:
Why don't you add default value?
```suggestion
pretrain_ngpu=1      # The number of gpus in pretrain stage ("0" uses cpu, otherwise use gpu).
pretrain_num_nodes=1 # The number of nodes in pretrain stage.
finetune_ngpu=1      # The number of gpus in finetune stage("0" uses cpu, otherwise use gpu).
finetune_num_nodes=1 # The number of nodes in finetune stage.
```